utils/resultwriter_test.go

```diff
@@ -441,3 +442,154 @@ func TestJSONMarshall(t *testing.T) {
 		})
 	}
 }
+
+func TestGetSummary(t *testing.T) {
+	dummyExtendedScanResults := &ExtendedScanResults{
+		ApplicabilityScanResults: []*sarif.Run{
+			sarifutils.CreateRunWithDummyResults(sarifutils.CreateDummyPassingResult("applic_CVE-2")).WithInvocations([]*sarif.Invocation{
+				sarif.NewInvocation().WithWorkingDirectory(sarif.NewSimpleArtifactLocation("target1")),
+			}),
+		},
+		SecretsScanResults: []*sarif.Run{
+			sarifutils.CreateRunWithDummyResults(sarifutils.CreateResultWithLocations("", "", "note", sarifutils.CreateLocation("target1/file", 0, 0, 0, 0, "snippet"))).WithInvocations([]*sarif.Invocation{
+				sarif.NewInvocation().WithWorkingDirectory(sarif.NewSimpleArtifactLocation("target1")),
+			}),
+			sarifutils.CreateRunWithDummyResults(sarifutils.CreateResultWithLocations("", "", "note", sarifutils.CreateLocation("target2/file", 0, 0, 0, 0, "snippet"))).WithInvocations([]*sarif.Invocation{
+				sarif.NewInvocation().WithWorkingDirectory(sarif.NewSimpleArtifactLocation("target2")),
+			}),
+		},
+		SastScanResults: []*sarif.Run{
+			sarifutils.CreateRunWithDummyResults(sarifutils.CreateResultWithLocations("", "", "note", sarifutils.CreateLocation("target1/file2", 0, 0, 0, 0, "snippet"))).WithInvocations([]*sarif.Invocation{
+				sarif.NewInvocation().WithWorkingDirectory(sarif.NewSimpleArtifactLocation("target1")),
+			}),
+		},
+	}
+
+	expectedVulnerabilities := &formats.ScanResultSummary{
+		ScaResults: &formats.ScaScanResultSummary{
+			ScanIds:      []string{TestScaScanId},
+			MoreInfoUrls: []string{TestMoreInfoUrl},
+			Security: formats.ResultSummary{
+				"Critical": map[string]int{jasutils.ApplicabilityUndetermined.String(): 1},
+				"High":     map[string]int{jasutils.NotApplicable.String(): 1},
+			},
+		},
+		SecretsResults: &formats.ResultSummary{"Low": map[string]int{jasutils.NotScanned.String(): 2}},
+		SastResults:    &formats.ResultSummary{"Low": map[string]int{jasutils.NotScanned.String(): 1}},
+	}
+	expectedViolations := &formats.ScanViolationsSummary{
+		Watches:   []string{"test-watch-name", "test-watch-name2"},
+		FailBuild: true,
+		ScanResultSummary: formats.ScanResultSummary{
+			ScaResults: &formats.ScaScanResultSummary{
+				ScanIds:      []string{TestScaScanId},
+				MoreInfoUrls: []string{TestMoreInfoUrl},
+				Security: formats.ResultSummary{
+					"Critical": map[string]int{jasutils.ApplicabilityUndetermined.String(): 1},
+					"High":     map[string]int{jasutils.NotApplicable.String(): 1},
+				},
+				License: formats.ResultSummary{"High": map[string]int{jasutils.NotScanned.String(): 1}},
+			},
+		},
+	}
+
+	testCases := []struct {
+		name                   string
+		results                *Results
+		includeVulnerabilities bool
+		includeViolations      bool
+		expected               formats.ResultsSummary
+	}{
+		{
+			name:                   "Vulnerabilities only",
+			includeVulnerabilities: true,
+			results: &Results{
+				ScaResults: []*ScaScanResult{{
+					Target:      "target1",
```

Comment:
try to set Target value with a more meaningful name

Reply:
most of those tests will be changed in the refactor